pkgs/initiator/initiator.go

```diff
@@ -1015,3 +1001,91 @@ func VerifySharesData(ops map[uint64]Operator, keys []*rsa.PrivateKey, ks *KeySh
 	}
 	return nil
 }
+
+func (c *Initiator) HealthCheck(ids []uint64) ([]*wire.Pong, error) {
+	ops, err := ValidatedOperatorData(ids, c.Operators)
+	if err != nil {
+		return nil, err
+	}
+	// Add messages verification coming form operators
+	verify, err := c.CreateVerifyFunc(ops)
+	if err != nil {
+		return nil, err
+	}
+	c.VerifyFunc = verify
```

Comment:
this is changed on other places? if yes should probably happen once on constructor.

Reply:
done